### ISSUE TITLE
Filtering with = operator on table clustered by property of object only returns subset of values inserted via UNNEST syntax

### ISSUE THREAD

**Issue by mtin** [crate/crate#10280]
**CrateDB version**: 4.1.8

**Environment description**: docker 19.03.8 with crate:4.1.8

**Problem description**:

The documentation recommends to use the `UNNEST` syntax for insertion as described [here](https://crate.io/docs/crate/howtos/en/latest/performance/inserts/methods.html).

When querying with a `WHERE` clause and the `=` operator on a table clustered by a property of an object for rows inserted with the `UNNEST` syntax only some of the rows in the table matching the comparison are returned, even though a full listing shows all objects have been inserted.

Querying using the `LIKE` operator returns all rows as expected. Same goes for the rows inserted with the value expression syntax. I created a minimal example that shows this behaviour:

**Steps to reproduce**:
Create table clustered on property of object:
```sql
CREATE TABLE IF NOT EXISTS crate.unnest_test(
    sample_object OBJECT(strict) as (
        some_id TEXT NOT NULL)
) CLUSTERED BY (sample_object['some_id']);
```
Insert rows with value expression:
```sql
INSERT INTO crate.unnest_test (sample_object) 
    VALUES (
        { some_id='indeedworking'}
    );
```

Insert rows with `UNNEST`:
```sql
INSERT INTO crate.unnest_test (sample_object)
    (SELECT *
        FROM UNNEST(
            [ { some_id='notworking'} ]
        )
    );
```

After inserting several rows with above inserts, when I query:
```sql
-- next line returns all elements inserted via value expression as expected
SELECT *  FROM crate.unnest_test WHERE sample_object['some_id']='indeedworking'; 
-- next line returns only some elements inserted via unnest <-- UNEXPECTED
SELECT *  FROM crate.unnest_test WHERE sample_object['some_id']='notworking';
-- returns all elements inserted via unnest
SELECT *  FROM crate.unnest_test WHERE sample_object['some_id'] LIKE 'notworking'; 
```

Could this be related to https://github.com/crate/crate/issues/10270? I cant try this on crate 4.2.1 because of that issue...

I am trying to wrap my head around this for a while but cant see why this should not work. Would be thankful for any advice. Cheers!

**Comment by kovrus** [comment]
@mtin We have just released CrateDB `4.2.2` testing version could you please try it out? The issue should be resolved there.

**Comment by mtin** [comment]
Can confirm this to be working with `4.2.2`. Thanks!
